fixtures/isolated-finish.html

```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <title>Test</title>
+    <script>
+      const oldOpen = window.open;
+      window.open = function() {
+        const win = oldOpen(...arguments);
+        const doc = win.document;
+        const scr = doc.createElement('script');
+        scr.innerHTML = `
+          Object.defineProperties(window, {
```

Comment:
We should add this script to the top page as well. Currently it would pass if we still did

```
// top level page
axe.finsinRun(results)
```

Reply:
I don't think this script works as we expect. Testing it out does not override the `finishRun` script due to how axe gets set up. 

In axe core, `axe` is added to `window` as [an empty object](https://github.com/dequelabs/axe-core/blob/develop/lib/core/index.js#L24), and then it adds the different public methods. So `finishRun` is added to `axe` and overrides the modified version, thus never throwing the error:

```html
<script>
Object.defineProperties(window, {
  _my_axe: {
    value: undefined,
    writable: true
  },
  axe: {
    get: function() {
      return this._my_axe;
    },
    set: function(axe) {
      this._my_axe = axe;
      axe.finishRun = () => { throw new Error('PWNED'); }
    }
  }
});
</script>
<script src="/axe.js"></script>
<script>
(async function() {
  const partial = await axe.runPartial();
  const results = await axe.finishRun([partial]);  // does not throw error
  console.log(results);
})();
</script>
```

Instead we need to listen for `finishRun` on the axe object or just on the `Object.prototype` itself (by abusing how JavaScript prototypes work). Even sneakier is we can add it to the `axe` object itself by deleting the defined property beforehand, making sure `axe.hasOwnProperty('finishRun') === true`:

```js
Object.defineProperty(Object.prototype, 'finishRun', {
  configurable: true,
  set(fn) {
   delete Object.prototype.finishRun;
   this.finishRun = function(partialResults) {
      throw new Error('PWNED');
    }
  }
});
```